modules/aruco/src/aruco.cpp

```diff
@@ -1713,20 +1719,19 @@ Ptr<GridBoard> GridBoard::create(int markersX, int markersY, float markerLength,
     }
 
     // calculate Board objPoints
-    float maxY = (float)markersY * markerLength + (markersY - 1) * markerSeparation;
     for(int y = 0; y < markersY; y++) {
         for(int x = 0; x < markersX; x++) {
-            vector< Point3f > corners;
-            corners.resize(4);
+            vector<Point3f> corners(4);
             corners[0] = Point3f(x * (markerLength + markerSeparation),
-                                 maxY - y * (markerLength + markerSeparation), 0);
+                                 y * (markerLength + markerSeparation), 0);
             corners[1] = corners[0] + Point3f(markerLength, 0, 0);
-            corners[2] = corners[0] + Point3f(markerLength, -markerLength, 0);
-            corners[3] = corners[0] + Point3f(0, -markerLength, 0);
+            corners[2] = corners[0] + Point3f(markerLength, markerLength, 0);
+            corners[3] = corners[0] + Point3f(0, markerLength, 0);
```

Comment:
You just changed the coordinate system by flipping Y.

Reply:
This changes was added to make coordinate systems consistent.
Now in `Board::create()` use this coordinate system:
![image](https://user-images.githubusercontent.com/22337800/154340852-5fdcfc6e-896b-4c7f-a5bc-3559f4d78e59.png)
But `Board::draw()` use another coordinate system:
![image](https://user-images.githubusercontent.com/22337800/154341321-d8f4908e-dffe-45d5-b80e-ac7cbb5da057.png)

Also, `Board::create()` set corners for marker in `objPoints` with "CCW" order:
![image](https://user-images.githubusercontent.com/22337800/154343015-3c2e7d0b-271c-41e0-b768-d0a421e5b678.png)
But `detectMarkers()` find markers and reordered corners to "CW" order:
![image](https://user-images.githubusercontent.com/22337800/154343485-e2cba76d-c4a3-4c68-9628-b728a94f793f.png)

This makes coordinate systems not consistent and makes the code hard to read